### PR TITLE
feat: (MenuLink) Allows for link to open in new window

### PR DESCRIPTION
**Pull request by MattSchiller**
### Overview

Allows callers to tell the link to open in a new window via `target`

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.39-alpha.ff5fcc.0<br/>@codecademy/gamut-labs@28.6.2-alpha.ff5fcc.0<br/>@codecademy/gamut@42.3.1-alpha.ff5fcc.0<br/>@codecademy/monaco@0.5.29-alpha.ff5fcc.0<br/>@codecademy/styleguide@52.5.1-alpha.ff5fcc.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d8bd7d6a3e78450469d475--gamut-preview.netlify.app>https://61d8bd7d6a3e78450469d475--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d8bd7d6a3e78450469d475>Deploy Logs</a></p>

File: packages/gamut/src/Menu/MenuItem.tsx [hunk 1]
```diff
@@ -17,49 +17,60 @@ export const MenuItem = React.forwardRef<
     'variant' | 'selected' | 'active-navlink'
   > & {
     href?: string;
+    openLinkInNewWindow?: boolean;
```
Comment (dreamwasp) [inline packages/gamut/src/Menu/MenuItem.tsx:20]: i wonder if we could future-proof this a bit more by making this a `Pick` of `_target` from `HTMLAnchorElement`?  extending from the DOM attribute feels a little straightforward to me then having a series of boolean props that do the same equivalent things

Comment (MattSchiller) [reply packages/gamut/src/Menu/MenuItem.tsx:20]: Absolutely my preference, too! I'd just thought we preferred to keep things "on rails", but lemme make this change 👍

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.39-alpha.2252ed.0<br/>@codecademy/gamut-labs@28.6.2-alpha.2252ed.0<br/>@codecademy/gamut@42.3.1-alpha.2252ed.0<br/>@codecademy/monaco@0.5.29-alpha.2252ed.0<br/>@codecademy/styleguide@52.5.1-alpha.2252ed.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61dc606661cf0de6e8819831--gamut-preview.netlify.app>https://61dc606661cf0de6e8819831--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61dc606661cf0de6e8819831>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.40-alpha.472e2b.0<br/>@codecademy/gamut-labs@28.6.3-alpha.472e2b.0<br/>@codecademy/gamut@42.3.1-alpha.472e2b.0<br/>@codecademy/monaco@0.5.29-alpha.472e2b.0<br/>@codecademy/styleguide@52.5.2-alpha.472e2b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61df3f10fdbe314b75116a42--gamut-preview.netlify.app>https://61df3f10fdbe314b75116a42--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61df3f10fdbe314b75116a42>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.46-alpha.854eda.0<br/>@codecademy/gamut-labs@29.0.1-alpha.854eda.0<br/>@codecademy/gamut@44.0.1-alpha.854eda.0<br/>@codecademy/monaco@0.6.1-alpha.854eda.0<br/>@codecademy/styleguide@54.0.1-alpha.854eda.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61f08b7bc37a4a157aa7555a--gamut-preview.netlify.app>https://61f08b7bc37a4a157aa7555a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61f08b7bc37a4a157aa7555a>Deploy Logs</a></p>

**Comment by MattSchiller** [comment]
@vivsnguyen Is this feature still needed? I haven't heard anything since this originally went up

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.57-alpha.969344.0<br/>@codecademy/gamut-labs@29.2.8-alpha.969344.0<br/>@codecademy/gamut@44.3.3-alpha.969344.0<br/>@codecademy/monaco@0.6.10-alpha.969344.0<br/>@codecademy/styleguide@54.1.3-alpha.969344.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61fd81afc5b9b3144284db35--gamut-preview.netlify.app>https://61fd81afc5b9b3144284db35--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61fd81afc5b9b3144284db35>Deploy Logs</a></p>

**Comment by MattSchiller** [comment]
Confirmed working in: https://github.com/codecademy-engineering/business-account-management/pull/566/files

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.58-alpha.005401.0<br/>@codecademy/gamut-labs@29.2.9-alpha.005401.0<br/>@codecademy/gamut@44.4.1-alpha.005401.0<br/>@codecademy/monaco@0.6.11-alpha.005401.0<br/>@codecademy/styleguide@54.2.1-alpha.005401.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61fd987d37ee9f008b7e94b6--gamut-preview.netlify.app>https://61fd987d37ee9f008b7e94b6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61fd987d37ee9f008b7e94b6>Deploy Logs</a></p>

**Pull request merged**